MapboxNavigation/NavigationMapView.swift

```diff
@@ -24,28 +24,16 @@ open class NavigationMapView: MGLMapView {
             return
         }
         
-        if let line = style.layer(withIdentifier: routeLayerIdentifier) {
-            style.removeLayer(line)
-        }
-        
-        if let lineCasing = style.layer(withIdentifier: routeLayerCasingIdentifier) {
-            style.removeLayer(lineCasing)
-        }
-        
-        if let source = style.source(withIdentifier: sourceIdentifier) {
-            style.removeSource(source)
-        }
-        
-        let polyline = shape(describing: route)
+        let polyline = navigationMapDelegate?.navigationMapView(self, shapeDescribing: route) ?? shape(describing: route)
```

Comment:
`NavigationMapView.shape(describing:)` etc. no longer need to be public, because the developer is expected to use the delegate methods instead.